test/integration/main.test.js

```diff
@@ -377,3 +377,216 @@ test('createRemoteBundle(sources) - other status codes', async () => {
     );
     await closeServer(server);
 });
+
+test('publishAssets(tag, files, options) - 400 error', async () => {
+    expect.assertions(1);
+    const { server, port } = await createTestServer([
+        {
+            verb: 'post',
+            path: '/publish-assets',
+            cb(req, res) {
+                res.status(400).send({ message: 'Bad request' });
+            },
+        },
+    ]);
+    const client = new Client({ buildServerUri: `http://127.0.0.1:${port}` });
+
+    const result = client.publishAssets('podlet1', ['first.js', 'second.js']);
+
+    await expect(result).rejects.toEqual(new Error('Bad request'));
+    await closeServer(server);
+});
+
+test('publishAssets(tag, files, options) - 500 error', async () => {
+    expect.assertions(1);
+    const { server, port } = await createTestServer([
+        {
+            verb: 'post',
+            path: '/publish-assets',
+            cb(req, res) {
+                res.status(500).send({ message: 'Server error' });
+            },
+        },
+    ]);
+    const client = new Client({ buildServerUri: `http://127.0.0.1:${port}` });
+
+    const result = client.publishAssets('podlet1', ['first.js', 'second.js']);
+
+    await expect(result).rejects.toMatchSnapshot();
+    await closeServer(server);
+});
+
+test('publishAssets(tag, files, options) - js', async () => {
+    expect.assertions(1);
+    const { server, port } = await createTestServer([
+        {
+            verb: 'post',
+            path: '/publish-assets',
+            cb(req, res) {
+                res.send(JSON.stringify(req.body));
+            },
+        },
+    ]);
+    const client = new Client({ buildServerUri: `http://127.0.0.1:${port}` });
+
+    const result = await client.publishAssets('podlet1', [
+        'first.js',
+        'second.js',
+    ]);
+
+    expect(result).toMatchSnapshot();
+    await closeServer(server);
+});
+
+test('publishAssets(tag, files, options) - uses plugins', async () => {
+    expect.assertions(1);
+    const { server, port } = await createTestServer([
+        {
+            verb: 'post',
+            path: '/publish-assets',
+            cb(req, res) {
+                res.send(JSON.stringify(req.body));
+            },
+        },
+    ]);
+    const client = new Client({ buildServerUri: `http://127.0.0.1:${port}` });
+    const fakePlugin = { _id: 10 };
+    const fakePluginOptions = { _id: 20 };
+    client.plugin(fakePlugin, fakePluginOptions);
+
+    await client.publishAssets('podlet1', ['first.js', 'second.js']);
+
+    expect(mockPlugin.mock.calls[0]).toEqual([fakePlugin, fakePluginOptions]);
```

Comment:
what's `mockPlugin`? Can't seems to find it...

Also, what about `toHaveBeenCalledWith`?

Reply:
top of the file: `const mockPlugin = jest.fn();`

Reply:
`toHaveBeenCalledWith` sounds good.